modules/REST/V1/Languages.php

```diff
@@ -0,0 +1,809 @@
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Languages REST controller.
+ *
+ * @since 3.7
+ */
+class Languages extends WP_REST_Controller {
+	/**
+	 * @var Languages_Model
+	 */
+	private $languages;
+
+	/**
+	 * @var PLL_Translatable_Objects
+	 */
+	private $translatable_objects;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since 3.7
+	 *
+	 * @param Languages_Model          $languages            Languages' model.
+	 * @param PLL_Translatable_Objects $translatable_objects Translatable objects registry.
+	 */
+	public function __construct( Languages_Model $languages, PLL_Translatable_Objects $translatable_objects ) {
+		$this->namespace            = 'pll/v1';
+		$this->rest_base            = 'languages';
+		$this->languages            = $languages;
+		$this->translatable_objects = $translatable_objects;
+	}
+
+	/**
+	 * Registers the routes for languages.
+	 *
+	 * @since 3.7
+	 *
+	 * @return void
+	 */
+	public function register_routes(): void {
+		register_rest_route(
+			$this->namespace,
+			"/{$this->rest_base}",
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_items' ),
+					'permission_callback' => array( $this, 'get_items_permissions_check' ),
+					'allow_batch'         => array( 'v1' => true ),
+				),
+				array(
+					'methods'             => WP_REST_Server::CREATABLE,
+					'callback'            => array( $this, 'create_item' ),
+					'permission_callback' => array( $this, 'create_item_permissions_check' ),
+					'allow_batch'         => array( 'v1' => true ),
+					'args'                => $this->get_endpoint_args_for_item_schema( WP_REST_Server::CREATABLE ),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+
+		register_rest_route(
+			$this->namespace,
+			"/{$this->rest_base}/(?P<term_id>[\d]+)",
+			array(
+				'args'   => array(
+					'term_id' => array(
+						'description' => __( 'Unique identifier for the language.', 'polylang' ),
+						'type'        => 'integer',
+					),
+				),
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_item' ),
+					'permission_callback' => array( $this, 'get_item_permissions_check' ),
+					'args'                => array(
+						'context' => $this->get_context_param( array( 'default' => 'view' ) ),
+					),
+				),
+				array(
+					'methods'             => WP_REST_Server::EDITABLE,
+					'callback'            => array( $this, 'update_item' ),
+					'permission_callback' => array( $this, 'update_item_permissions_check' ),
+					'args'                => $this->get_endpoint_args_for_item_schema( WP_REST_Server::EDITABLE ),
+				),
+				array(
+					'methods'             => WP_REST_Server::DELETABLE,
+					'callback'            => array( $this, 'delete_item' ),
+					'permission_callback' => array( $this, 'delete_item_permissions_check' ),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
```

Comment:
`allow_batch` could be added here.

Reply:
Done in [97f252f](https://github.com/polylang/polylang/pull/1505/commits/97f252fe332b2ce1b268b1286e05a17022e14184).